src/mtype.d

```diff
@@ -7536,12 +7536,21 @@ public:
         }
         if (exp.op == TOKscope)
         {
-            ScopeDsymbol sds = (cast(ScopeExp)exp).sds;
+            auto sds = (cast(ScopeExp)exp).sds;
             if (sds.isPackage())
             {
                 error(loc, "%s has no type", exp.toChars());
                 goto Lerr;
             }
+            if (auto ti = sds.isTemplateInstance())
+            {
+                if (ti.needsTypeInference(sc))
+                {
+                    // Bugzilla 15550: ti is a partial instantiattion form fun!tiargs.
```

Comment:
Fixed typo,

The suggested message would be better, but diagnostic improvement is beyond the purpose of this PR.

I have a plan to move the "has no type" message into `ScopeExp.checkValue()`.